### PR TITLE
[RW-9762][risk=no] Refactor: Convert GKE apps HOC to a typical component

### PR DESCRIPTION
**Pull request by Peter-Lavigne**
Converts GKE apps HOC to a typical component.

In a past PR, I had created an HOC to share code between the RStudio and Cromwell config panels because I was trying to avoid creating a complex component that rendered both RStudio and Cromwell. While working on adding persistent disk support, I realized I can achieve the same result using a parent component. I think this approach is easier to understand and makes adding another API call straightforward.

There are no user-facing changes in this PR.

I manually tested this by opening and closing the GKE app panels and observing that the panels use the data from the "list apps" API call.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 1]
```diff
@@ -701,12 +700,12 @@ describe('HelpSidebar', () => {
 
     expect(wrapper.text()).not.toContain('RStudio Cloud Environment');
 
-    const cromwellIcon = wrapper.find({
```
Comment (Peter-Lavigne) [inline ui/src/app/components/help-sidebar.spec.tsx:704]: `cromwellIcon` => `rstudioIcon` since this is an RStudio test. This is unrelated to the PR.

File: ui/src/app/components/rstudio-configuration-panel.spec.tsx [hunk 2]
```diff
@@ -22,39 +22,38 @@ import {
...
+    cdrVersionId: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID,
+  },
+  profileState: {
+    profile: ProfileStubVariables.PROFILE_STUB,
+    load: jest.fn(),
+    reload: jest.fn(),
+    updateCache: jest.fn(),
+  },
+  gkeAppsInWorkspace: [],
+};
```
Comment (Peter-Lavigne) [inline ui/src/app/components/rstudio-configuration-panel.spec.tsx:47]: This is the same code as before but moved out of the `describe` block and `export`ed for use in `gke-app-configuration-panel.spec.tsx`.

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    });
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(listAppsStub).toHaveBeenCalledWith(workspaceNamespace);
+    const cromwellPanel = wrapper.find(CromwellConfigurationPanel);
+    expect(cromwellPanel.exists()).toBeTruthy();
+    expect(cromwellPanel.prop('gkeAppsInWorkspace')).toEqual(apps);
+  });
+
+  it('should display the Cromwell panel when the type is Cromwell', async () => {
```
Comment (Peter-Lavigne) [inline ui/src/app/components/gke-app-configuration-panel.spec.tsx:115]: This test and the RStudio test are both new. We weren't testing the `cond` statement before.

**Review by chenchals — APPROVED** [review]
Just FYI - react is not my forte...

**Review by yonghaoy — APPROVED** [review]
